internal/topology/variables/cluster_variable_defaulting.go

```diff
@@ -86,83 +83,141 @@ func defaultClusterVariables(clusterVariables []clusterv1.ClusterVariable, clust
 		// Continue if there is no defaulted variable.
 		// NOTE: This happens when the variable doesn't exist on the CLuster before and
 		// there is no top-level default value.
-		if defaultedClusterVariable == nil {
+		if defaultedValue == nil {
 			continue
 		}
-
-		defaultedClusterVariables = append(defaultedClusterVariables, *defaultedClusterVariable)
+		defaultedValues = append(defaultedValues, *defaultedValue)
 	}
 
 	if len(allErrs) > 0 {
 		return nil, allErrs
 	}
+	return defaultedValues, nil
+}
 
-	return defaultedClusterVariables, nil
+// getCurrentValue returns the value of a variable for its definitionFrom, or for an empty definitionFrom if it exists.
+func getCurrentValue(variable clusterv1.ClusterVariable, valuesMap map[string]map[string]clusterv1.ClusterVariable) *clusterv1.ClusterVariable {
+	// If the value is set in the Cluster spec get the value.
+	if valuesForName, ok := valuesMap[variable.Name]; ok {
+		if value, ok := valuesForName[variable.DefinitionFrom]; ok {
+			return &value
+		}
+	}
+	return nil
 }
 
-// defaultClusterVariable defaults a clusterVariable based on the default value in the clusterClassVariable.
-func defaultClusterVariable(clusterVariable *clusterv1.ClusterVariable, clusterClassVariable *clusterv1.ClusterClassVariable, fldPath *field.Path, createVariable bool) (*clusterv1.ClusterVariable, field.ErrorList) {
-	if clusterVariable == nil {
+// defaultValue defaults a clusterVariable based on the default value in the clusterClassVariable.
+func defaultValue(currentValue *clusterv1.ClusterVariable, definition *statusVariableDefinition, fldPath *field.Path, createVariable bool) (*clusterv1.ClusterVariable, field.ErrorList) {
```

Comment:
kudos for cleaning up the wording (the distinction between value/definition make code more readable)

Reply:
Absolutely agree. clusterVariable vs clusterClassVariable was driving me slightly crazy :)
(disclaimer I wrote this code initially)